test/unit/specs/core/edgeNetwork/sendEdgeNetworkRequestFactory.spec.js

```diff
@@ -414,14 +414,14 @@ describe("sendEdgeNetworkRequestFactory", () => {
 
   it("returns the merged object from  lifecycle::onBeforeRequest & lifecycle::onResponse", () => {
     lifecycle.onBeforeRequest.and.callFake(({ onResponse }) => {
-      onResponse(() => [{ a: 1 }, undefined, [], {}]);
+      onResponse(() => [{ a: 1 }, { b: 1 }, undefined]);
```

Comment:
I wouldn't expect a component to return an array of objects from the `onResponse` callback. I'm either misunderstanding (very likely) or something is wrong.

Reply:
I took a deeper look. The callback passed into `onResponse` should be returning a single object only. This test should really be failing as it is.

If you're trying to test that objects returned from the `onResponse` callbacks from multiple components get merged together properly, you would do something like this instead:

```
lifecycle.onBeforeRequest.and.callFake(({ onResponse }) => {
  onResponse(() => ({ a: 1 }));
  onResponse(() => ({ b: 1 })]);
  onResponse(() => undefined]);
});
```

Reply:
thank you, I'll fix my tests with your suggestions